Code/GraphMol/MolHash/nmmolhash.h

```diff
@@ -0,0 +1,59 @@
+/*==============================================*/
+/* Copyright (C)  2019       NextMove Software  */
+/* All rights reserved.                         */
+/*                                              */
+/* This file is part of molhash.                */
+/*                                              */
+/* The contents are covered by the terms of the */
+/* BSD license, which is included in the file   */
+/* license.txt.                                 */
+/*==============================================*/
+
+// This file will disappear in a future release; please don't include it
+// directly. Use MolHash.h instead.
+
+#include <RDGeneral/export.h>
+#ifndef NMS_MOLHASH_H
+#define NMS_MOLHASH_H
+
+#include <string>
+#include <vector>
+
+namespace RDKit {
+class RWMol;
+namespace MolHash {
+enum class RDKIT_MOLHASH_EXPORT HashFunction {
+  AnonymousGraph = 1,
+  ElementGraph = 2,
+  CanonicalSmiles = 3,
+  MurckoScaffold = 4,
+  ExtendedMurcko = 5,
+  MolFormula = 6,
+  AtomBondCounts = 7,
+  DegreeVector = 8,
+  Mesomer = 9,
+  HetAtomTautomer = 10,
+  HetAtomProtomer = 11,
+  RedoxPair = 12,
+  Regioisomer = 13,
+  NetCharge = 14,
+  SmallWorldIndexBR = 15,
+  SmallWorldIndexBRL = 16,
+  ArthorSubstructureOrder = 17
+};
+
+RDKIT_MOLHASH_EXPORT std::string MolHash(RWMol *mol, enum HashFunction func);
+
+enum class RDKIT_MOLHASH_EXPORT StripType {
+  AtomStereo = 1,
+  BondStereo = 2,
+  Isotope = 4,
+  AtomMap = 8,
+  Hydrogen = 16
+};
+
+RDKIT_MOLHASH_EXPORT void Strip(RWMol *mol, unsigned int striptype);
+RDKIT_MOLHASH_EXPORT void SplitMolecule(RWMol *mol, std::vector<RWMol *> &molv);
```

Comment:
Unless we want to actually allow molv to be an accumulator, i prefer

 `std::vector<ROMol*> SplitMolecule(mol)`

There is no overhead for returning a vector.

Reply:
Or maybe

`std::vector<unique_ptr<ROMol>> SplitMolecule(mol)`

:)

Reply:
I understand your point, but there are many examples in the RDKit API that, for historic reasons, take the vector as an argument. So the current form isn't really inconsistent.

It's probably worth making a decision some point about how we think it's best to handle this and write that down somewhere, but we haven't done that yet.

Reply:
I agree in general.  The real deciding factor for me is that it would unify the python wrappers and C++.  I.e. the combination of unique_ptr and returning vectors would make the APIs identical except for capitalization.

Reply:
Yeah, and it would probably remove the need for a helper function to do the wrapper.
Well, except for the unique_ptrs... you can't give those to Python (need shared pointers for that).